### PR TITLE
Bump to Tokio 0.3.0 and clean up a few dependencies

### PR DESCRIPTION
**Pull request by udoprog**
Not a ton of changes this release. Definitely not as many as `0.2` 🎉 

Removed dependencies:
* `futures-channel`
* `pin-utils` - seems to no longer used, and tokio comes with [tokio::pin](https://docs.rs/tokio/0/tokio/macro.pin.html) now.

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -95,326 +95,338 @@ pub mod transport;
 
 macro_rules! pub_state_base {
     () => {
```
Comment (udoprog) [inline src/client/mod.rs:97]: These changes are mostly done by rustfmt.

File: Cargo.toml [hunk 2]
```diff
@@ -36,24 +36,22 @@ yaml = ["yaml_config"]
...
 
-tls-native = ["native-tls", "tokio-tls"]
+tls-native = ["native-tls", "tokio-native-tls"]
 tls-rust = ["tokio-rustls", "webpki-roots"]
 
 
 [dependencies]
 bufstream = "0.1.0"
-bytes = "0.5.0"
+bytes = "0.6.0"
```
Comment (tirz) [inline Cargo.toml]: `bufstream` and `bytes` dependencies are not used by the crate.
I guess we can directly removed them.

**Comment by freddyb** [comment]
@tirz, @udoprog: Do you intend to continue with this PR? If not, I'm happy to pick it up.
I am hoping to get a tokio-0.3-based irc crate soon :)

**Comment by udoprog** [comment]
@freddyb yeah. I was waiting for word from @aatxe. I forgot there was one dep that could be removed but figured this was not a blocker.

**Comment by aatxe** [comment]
Sorry, I think I got blocked on reviewing it because of all the formatting changes, and then failed to get back to it. It looks fine, and I'm happy to accept it. Can you rebase (and feel free to remove the other dependency too)?

**Comment by udoprog** [comment]
Rebased and put formatting in a single commit at the head.

`bytes` and `bufstream` deps are gone.

Github easily lets you filter out the commits you want to see, so the relevant diff can be viewed like this:
https://github.com/aatxe/irc/pull/220/files/56e584fe76f788c3430ef1e56a4c795d03946e25..370fb1c0885830e593e78fdc020015c9ee9587eb

**Comment by aatxe** [comment]
Looks good to me, cheers!

**Comment by freddyb** [comment]
Thank you, you fine people :rainbow:

**Pull request merged**